File: pubnub-gson/src/main/java/com/pubnub/api/endpoints/message_actions/AddMessageAction.java
package com.pubnub.api.endpoints.message_actions;

import com.pubnub.api.endpoints.Endpoint;
import com.pubnub.api.models.consumer.message_actions.PNAddMessageActionResult;
import com.pubnub.api.models.consumer.message_actions.PNMessageAction;

public interface AddMessageAction extends Endpoint<PNAddMessageActionResult> {
    AddMessageAction channel(String channel);

    AddMessageAction messageAction(PNMessageAction messageAction);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/message_actions/AddMessageAction.kt
package com.pubnub.api.endpoints.message_actions

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.message_actions.PNAddMessageActionResult
import com.pubnub.api.models.consumer.message_actions.PNMessageAction

/**
 * @see [PubNub.addMessageAction]
 */
interface AddMessageAction : com.pubnub.kmp.endpoints.message_actions.AddMessageAction, Endpoint<PNAddMessageActionResult> {
    val channel: String
    val messageAction: PNMessageAction
}
